### PR TITLE
add --cluster and --context flag support to login

### PR DESCRIPTION
**Pull request by juanvallejo**
Related bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1419857

This patch checks the values of these two flags and updates the
current context in kubeconfig based on that value.

cc @openshift/cli-review @deads2k

### CODE REVIEW

File: pkg/cmd/cli/cmd/login/login.go [hunk 1]
```diff
@@ -192,6 +193,30 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
...
 func RunLogin(cmd *cobra.Command, options *LoginOptions) error {
+	contextName := kcmdutil.GetFlagString(cmd, "context")
+	if len(contextName) > 0 {
+		if _, exists := options.StartingKubeConfig.Contexts[contextName]; exists {
+			options.StartingKubeConfig.CurrentContext = contextName
+			if err := kclientcmd.ModifyConfig(options.PathOptions, *options.StartingKubeConfig, true); err != nil {
+				return err
+			}
+			return nil
+		}
```
Comment (juanvallejo) [inline pkg/cmd/cli/cmd/login/login.go]: If the context provided is not valid, then the command either attempts to use the value from `--cluster`, if one was give, or resumes normal execution. Not sure if that is the right approach, or if an error should be returned here instead.

**Comment by juanvallejo** [comment]
[test]

**Comment by juanvallejo** [comment]
cc @deads2k or @liggitt

**Comment by liggitt** [comment]
login builds a non-conflicting context name from the combination of the server and user... what is the use case for manually managing the context the login applies to?

**Comment by juanvallejo** [comment]
@liggitt The way I interpreted the linked bugzilla is that the user is using `oc login` to switch between existing contexts, and expects the command to perform the behavior of `oc set-contex ...` / `oc set-cluster ...` before the "login" behavior if the `--cluster` or `--context` option is passed. If this is something that would not be considered acceptable behavior for login, maybe we could output a warning when those flags are used, explaining that they're not supposed to mutate the config, but are rather a single-instance behavior?

**Comment by liggitt** [comment]
`oc project` will switch between contexts, I don't think we want to start passing context names to login

**Comment by juanvallejo** [comment]
@liggitt sounds good, I can go ahead and add a warning if a user tries to use these flags with `oc login`, if that is okay. We could suggest `oc project ...` or `oc config set-context ...` in case that is their intent when doing so

**Pull request closed without merging**